latex.bzl

```diff
@@ -52,34 +75,36 @@ _latex_pdf = rule(
             default = "@bazel_latex_latexrun//:latexrun",
         ),
     },
-    outputs = {"out": "%{name}.pdf"},
+    outputs = {"out": "%{name}.%{format}"},
     toolchains = ["@bazel_latex//:latex_toolchain_type"],
-    implementation = _latex_pdf_impl,
+    implementation = _latex_impl,
 )
 
-def latex_document(name, main, srcs = [], tags = [], cmd_flags = []):
-    # PDF generation.
-    _latex_pdf(
+def latex_document(name, main, srcs = [], tags = [], cmd_flags = [], format="pdf"):
+
+    _latex(
         name = name,
         srcs = srcs + ["@bazel_latex//:core_dependencies"],
         main = main,
         tags = tags,
         cmd_flags = cmd_flags,
+        format = format,
     )
 
-    # Convenience rule for viewing PDFs.
-    native.sh_binary(
-        name = name + "_view_output",
-        srcs = ["@bazel_latex//:view_pdf.sh"],
-        data = [name + ".pdf"],
-        tags = tags,
-    )
+    if "pdf" in format:
```

Comment:
I'm really not familiar with the dvi format - is there an easy way to view such a file?

Reply:
I looked into the shell script and it does xdg-open on linux, open on osx,
so in that way there is no need to have the if-statement in the macro.
However, the script it self searches only for ".pdf" files from what I can see.

other than that, I think that if the OS have been told that some program can open dvi files, it will use that program, that is how I recall the xdg-open works.

The most correct way, in my opinion, would be to send in the latex_output as input to the args, something like,
         native.sh_binary(
             name = "{}_view_output".format(name),
             srcs = ["@bazel_latex//:view_pdf.sh"],
             data = [":{}".format(name)],
             args = ["$(location :{})".format(name)]
             tags = tags,
         )

Then we would't have to make use of 'find' at all.
I'd say we could open another issue for that?

Reply:
Yep that sounds good to me! I'll take that one at some point this week